spyderlib/plugins/ipythonconsole.py

```diff
@@ -1058,8 +1058,6 @@ def _create_client_for_kernel(self, cf, hostname, kf, pw):
         # Verifying if the connection file exists
         cf = osp.basename(cf)
         try:
-            if not cf.startswith('kernel') and not cf.endswith('json'):
-                cf = to_text_string('kernel-' + cf + '.json')

```

Comment:
The idea is to pass the kernel id (like just `4779`) to this field and be able to connect to that kernel, instead of writing `kernel-4779.json`.

That's why these lines are here and I find this feature really useful :-)